include/maths/common/CLeastSquaresOnlineRegressionDetail.h

```diff
@@ -216,6 +216,11 @@ bool CLeastSquaresOnlineRegression<N, T>::parameters(std::size_t n,
 
     typename SJacobiSvd<MATRIX>::Type svd(x.template selfadjointView<Eigen::Upper>(),
                                           Eigen::ComputeFullU | Eigen::ComputeFullV);
+    if (svd.info() != Eigen::Success) {
```

Comment:
If this fixes the crash then the same protection is needed on line 252, which also calls `selfadjointView`.

Also line 674 of `CLinearAlgebraEigen.h` and line 174 of `CBoostedTreeLeafNodeStatisticsThreading.cc`.

Reply:
It's only singular value decomposition (SVD) matrices that potentially have the issue with undefined behaviour upon construction (and indeed have the `info()` method to check success or not) so it's only possible to implement the first suggested change here, in `include/maths/common/CLeastSquaresOnlineRegressionDetail.h` line 252